tests/run_tests.py

```diff
@@ -351,28 +355,38 @@ def run_system(config_file, vic_exe, test_data_dir, out_dir, driver):
                                                         expectation))
 
             # -------------------------------------------------------- #
-            # check output files
+            # check output files (different tests depending on driver)
             if test_dict['check']:
-                if 'complete' in test_dict['check']:
-                    start = None
-                    end = None
-
-                for fname in glob.glob(os.path.join(dirs['results'], '*')):
-                    df = read_vic_ascii(fname, header=True)
-                    # do numeric tests
-
-                    # check that each dataframe includes all timestamps
                     if 'complete' in test_dict['check']:
-                        if (start is not None) and (end is not None):
-                            check_completed(df, start, end)
-                        else:
-                            start = df.index[0]
-                            end = df.index[-1]
-
-                    # check for nans in the df
-                    if 'nonans' in test_dict['check']:
-                        check_for_nans(df)
-            # -------------------------------------------------------- #
+                        start = None
+                        end = None
+                        
+                        if driver == "classic": 
+                    
+                            for fname in glob.glob(os.path.join(dirs['results'], '*')):
+                                df = read_vic_ascii(fname, header=True)
+                                # do numeric tests
+
+                                # check that each dataframe includes all timestamps
+                                if (start is not None) and (end is not None):
+                                    check_completed(df, start, end)
+                                else:
+                                    start = df.index[0]
+                                    end = df.index[-1]
+
+                    if 'output_file_nans' in test_dict['check']:
+
+                        if driver == "classic": 
+                            # check for nans in the df 
+                            check_for_nans(df) 
+
+                        elif driver == "image": 
+                            # check for nans in domain file  
+                            ds_domain = xr.open_dataset(os.path.join(test_dir, 'system', 
+                                                        test_dict['domain_file']))
+                            ds_output = xr.open_dataset(os.path.join(dirs['results'], 
+                                                        'Stehekin.history.nc')) 
```

Comment:
Instead of hard-coding this file name, can we do a `glob` wildcard (`*.nc`) inside the results directory and check all files present?